### PR TITLE
Add new Filename Generator with date, node, and filter

### PR DESCRIPTION
**Pull request by cfkoehler**
This is similar to https://github.com/NationalSecurityAgency/emissary/blob/master/src/main/java/emissary/output/io/DateStampFilenameGenerator.java

But instead of trying to update that class to what we need figured I would create a new one. I can deprecate the old one if we want. Or just leave it alone

### CODE REVIEW

File: src/test/java/emissary/output/io/DateFilterFilenameGeneratorTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import org.apache.commons.lang3.math.NumberUtils;
+import org.junit.jupiter.api.Test;
+
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class DateFilterFilenameGeneratorTest extends UnitTest {
```
Comment (fbruton) [inline src/test/java/emissary/output/io/DateFilterFilenameGeneratorTest.java:17]:
```suggestion
class DateFilterFilenameGeneratorTest extends UnitTest {
```

File: src/main/java/emissary/output/io/DateFilterFilenameGenerator.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.UUID;
+
+public class DateFilterFilenameGenerator implements FileNameGenerator {
+
+    protected static final SimpleDateFormat DATE_PATTERN = new SimpleDateFormat("yyyyDDDHHmm");
```
Comment (fbruton) [inline src/main/java/emissary/output/io/DateFilterFilenameGenerator.java:14]:
```suggestion
    protected static final DateTimeFormatter DATE_PATTERN = DateTimeFormatter.ofPattern("yyyyDDDHHmm");
```

File: src/main/java/emissary/output/io/DateFilterFilenameGenerator.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    public String nextFileName() {
+        return createFileName(filterNamePart);
+    }
+
+    public static String createFileName(String extension) {
+        return String.format("%s%s%s%s%s%s", now(), DELIMITER, UUID.randomUUID(), DELIMITER, NODE_NAME, extension);
+    }
+
+    private static String now() {
+        return DATE_PATTERN.format(new Date());
```
Comment (fbruton) [inline src/main/java/emissary/output/io/DateFilterFilenameGenerator.java:43]:
```suggestion
        return DATE_PATTERN.format(LocalDateTime.now());
```

**Review by fbruton — CHANGES_REQUESTED** [review]

File: src/main/java/emissary/output/io/DateFilterFilenameGenerator.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import emissary.directory.EmissaryNode;
+import emissary.util.io.FileNameGenerator;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.UUID;
+
+public class DateFilterFilenameGenerator implements FileNameGenerator {
```
Comment (drivenflywheel) [inline src/main/java/emissary/output/io/DateFilterFilenameGenerator.java:12]: Don't really love this class name, but it'll do for now.  Should describe the generated filename _(maybe with an example?)_ in a Javadoc comment on the class itself or constructor.

**Review by drivenflywheel — CHANGES_REQUESTED** [review]
Minor Javadoc cleanup

File: src/main/java/emissary/output/io/DateFilterFilenameGenerator.java (hunk 4 above)
Comment (cfkoehler) [reply src/main/java/emissary/output/io/DateFilterFilenameGenerator.java:12]: I am stuck on a better name. Unless it was going to be really long to be descriptive.
But definitely will add a java doc

**Review by fbruton — APPROVED** [review]

File: src/main/java/emissary/output/io/DateFilterFilenameGenerator.java (hunk 4 above)
Comment (drivenflywheel) [reply src/main/java/emissary/output/io/DateFilterFilenameGenerator.java:16]: Again, not in scope for this PR, but we could more filename generators to a child package and then remove the redundant `FilenameGenerator` suffix so the names can have sufficient info without being entire sentences. Alternatively, we could invert the generators so they take a provided NamingStrategy.  (Not sure that last one will always work.)

**Review by drivenflywheel — APPROVED** [review]

**Pull request merged**